grakn-engine/src/main/java/ai/grakn/engine/controller/TasksController.java

```diff
@@ -210,23 +209,24 @@ private Json createTask(Request request, Response response) {
     private Class<?> getClass(String className){
         try {
             Class<?> clazz = Class.forName(className);
-            if (!BackgroundTask.class.isAssignableFrom(clazz)) {
-                throw new GraknEngineServerException(400, ErrorMessage.UNAVAILABLE_TASK_CLASS, className);
-            }
-
+            if (!BackgroundTask.class.isAssignableFrom(clazz)) throw GraknServerException.invalidTask(className);
             return clazz;
         } catch (ClassNotFoundException e) {
-            throw new GraknEngineServerException(400, ErrorMessage.UNAVAILABLE_TASK_CLASS, className);
+            throw GraknServerException.invalidTask(className);
         }
     }
 
     /**
      * Error accessing or retrieving a task from storage. This throws a 404 Task Not Found to the user.
-     * @param exception {@link EngineStorageException} thrown by the server
+     * @param exception {@link GraknServerException} thrown by the server
      * @param response The response object providing functionality for modifying the response
      */
     private void handleNotFoundInStorage(Exception exception, Response response){
-        response.status(404);
+        if(exception instanceof GraknServerException){
```

Comment:
Basically the problem is that `GraknServerException extends GraknBackendException `